### ISSUE TITLE
Schema with no viewable/editable/insertable field will crash

### ISSUE THREAD

**Issue by eric-burel** [VulcanJS/Vulcan#1905]
Hi,

A schema without any `viewableBy`, `editableBy` or `insertableBy` will make the app crash when creating the corresponding collection, saying that `Type YourTypeName is not defined in document` (meaning that the corresponding gql type do not exist I guess).

I don't know if there is any use case for a schema like this, maybe defining a collection meant to be manipulated by other collections (eg some aggregated companion object).
I suspect that the app should not crash even if it is not a "normal usage" so I prefer to raise the issue.

To reproduce:
```js
const Customers = createCollection({
  collectionName: 'Customers',
  typeName: 'Customer',
  schema: {
    _id: {
      type : String,
      optional: true,
    },
    name: {
      type: String,
      label: 'Name',
    }
  },
  resolvers: getDefaultResolvers('Customers'),
  mutations: getDefaultMutations('Customers'),
});

**Comment by SachaG** [comment]
On one hand I agree it shouldn't crash, but then again if you're adding GraphQL resolvers and mutations to a collection that has no GraphQL type (since the type would be empty), I'm not sure what's supposed to happen…?

**Comment by SachaG** [comment]
Or should we support types with no fields at all? Is that ever useful?

**Comment by eric-burel** [comment]
I agree it does not make sense, but if its valid I guess the best fit is a warning message from Vulcan.
Mongo, Apollo and Meteor are experts in terse, undocumented and unhelpful error messages, I spend sometimes dozen of minutes to figure out what I actually did wrong.

Of course I understand that adding such warnings also mean a lot more work for you and the contributors, this is not at all an emergency.

**Comment by SachaG** [comment]
In this case I feel like the error message is fairly explicit though: you're trying to use a type that does not exist in your GraphQL schema. But yeah I see the point, it might not be obvious what the root cause is. Let's close this for now and see if anybody else runs into the same issue, if it turns out to be a common problem we'll think about how to handle it more gracefully.

**Comment by Apollinaire** [comment]
I ran into this issue two weeks ago and I struggled to understand why the type does not exist in the schema. My problem was that I was just lazy to fill the viewableBy fields, because I wanted to prototype fast. When I saw the error message, I did not understand it, and i had to go through the movies example tutorial to understand that it came from the missing fields. 

For now, maybe just add a warning in the [Schema docs](http://docs.vulcanjs.org/schemas.html) ? Something like "If your schema has no `viewableBy` fields, the type will be empty and the app will crash" within the Example section or the Creating collections section?

**Comment by SachaG** [comment]
OK how about this: if you've specified a type but don't have any viewableBy/editableBy/etc. we show a warning.

**Comment by SachaG** [comment]
Happy to accept a PR to add that warning :)

**Comment by okanthony** [comment]
@SachaG I'm new to open source - would you consider this a good first issue?

**Comment by SachaG** [comment]
It's a simple enough task, but it does require understanding where various things are happening. So I'd say if you're interested in learning more about the inner workings of Vulcan it's a good issue, but if you'd like to work on something more targeted then maybe not?

**Comment by okanthony** [comment]
I'm definitely interested in learning more about Vulcan - will take a look, thanks!

**Comment by rickynguyen4590** [comment]
I ran into this issue and it was took me 4 hours. I think if you add a warnings on it..it will help me a lots. Because I am new on Vulcan, Graphql. Btw thank for your kind support.

**Comment by SachaG** [comment]
Actually there's a generateGraphQLSchema = false option you can pass to
explicitly disable schema generation so I think we can show a warning after
all.
